### ISSUE TITLE
Parser panics with malformed `if`

### ISSUE THREAD

**Issue by jlapeyre** [Qiskit/openqasm3_parser#104]
This source file
```
if
```
causes panic. The relevant part of the stack trace is
```
   3: oq3_parser::parser::Parser::bump
             at ./crates/oq3_parser/src/parser.rs:188:9
   4: oq3_parser::grammar::expressions::atom::block_expr
             at ./crates/oq3_parser/src/grammar/expressions/atom.rs:375:5
   5: oq3_parser::grammar::items::if_stmt
             at ./crates/oq3_parser/src/grammar/items.rs:126:5
```

**Comment by jlapeyre** [comment]
Remember to include version/branch/commit in bug report!

**Comment by jlapeyre** [comment]
* Inadvertently fixed in #109, i.e. it was not known to be fixed at the time.
